packages/visx-grid/src/types.ts

```diff
@@ -10,7 +10,11 @@ export type GridScale<Output extends GridScaleOutput = GridScaleOutput> =
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   D3Scale<Output, any, any>;
 
-export type GridLines = { from: { x?: number; y?: number }; to: { x?: number; y?: number } }[];
+export type GridLines = {
+  from: { x?: number; y?: number };
+  to: { x?: number; y?: number };
+  index: number;
```

Comment:
`AnimatedGridRows/Columns` in `@visx/react-spring` (where the bug occurs) extends `@visx/grid`, so we need to pass `index` through in this package.